pilot/pkg/model/push_context.go

```diff
@@ -230,7 +230,7 @@ func (first *PushRequest) Merge(other *PushRequest) *PushRequest {
 	}
 
 	// Merge the target namespaces
-	if len(first.NamespacesUpdated) > 0 || len(other.NamespacesUpdated) > 0 {
+	if len(first.NamespacesUpdated) > 0 && len(other.NamespacesUpdated) > 0 {
```

Comment:
I know I did this change in #17479 in my version. Looking at the logic again, shouldn't we be merging though? the list of namespaces to update, the list of configs to update, etc. ?

Reply:
I think this is correct actually. What we really want to represent here is `AllNamespaces | ListOfNamespaces`, but go makes this pretty hard to do so we instead model `AllNamespaces` as `nil`. So if either one is nil (all namespaces) we want to send to nil (all namespaces)?

Reply:
yes, think of the case a virtualservice triggered push and a service triggered push merge. We expect all namespaces updated.